app/form_models/referees/reference_information_form.rb

```diff
@@ -2,22 +2,22 @@
 
 module Referees
   class ReferenceInformationForm < ReferenceForm
-    FIELDS = %i[
-      under_investigation
-      warnings
-      allegations
-      not_fit_to_practice
-      able_to_undertake_role
-    ].freeze
-
-    FIELDS.each do |field_name|
+    JobReference::REFERENCE_INFO_FIELDS.each do |field_name|
       attribute field_name, :boolean
       validates field_name, inclusion: { in: [true, false], allow_nil: false }
     end
 
+    JobReference::REASON_FIELDS.each do |field_name|
+      attribute field_name, :string
+    end
+
+    validates :under_investigation_details, presence: true, if: -> { under_investigation }
+    validates :warning_details, presence: true, if: -> { warnings }
+    validates :unable_to_undertake_reason, presence: true, unless: -> { able_to_undertake_role }
```

Comment:
For consistency, shouldn't these validations also include the length validation as done in the `RefereeDetailsForm` ?

Reply:
They were validated, but I think the names were a little confusing. Hopefully they now have better names